### PR TITLE
Fixes #16 - Double check the UA string before proceeding

### LINKED ISSUE #16: Retrofill UA only if the user agent is Chrome
## Actual Behavior
Several [bots / spiders](https://www.iab.com/guidelines/iab-abc-international-spiders-bots-list/) likely use Chromium based browsers for automated web requests. Such agents right now identify themselves by modifying the User-Agent header. When client hints are introduced and a client side JS adopts uach-retrofill the automated requests from such user agents will contain Chrome User-Agent instead of the UA string of the bot / spider.

## Expected Behavior
Websites should continue to be able to identify bots using User-Agent even after using uach-retrofill.

## Suggested Change
Verify that `navigator.userAgent` identifies itself as Chrome before filling `navigator.userAgent` with a string constructed from client hints.

### PR DESCRIPTION
**Pull request by miketaylr**
PTAL @rowan-m

### CODE REVIEW

**Comment by miketaylr** [comment]
Ping @rowan-m

File: uach-retrofill.js [hunk 1]
```diff
@@ -105,7 +108,7 @@ async function getUserAgentUsingClientHints(hints) {
     }
   });
   if (!is_chromium) {
-    // If this is not a Chromium based browser, the UA string should be very different. So bailing...
+    // If this is not a Chromium-based browser, the UA string should be very different. So bailing...
```
[the remark this answers is not in the document]
Comment (miketaylr) [reply uach-retrofill.js:111]: 🐛

File: uach-retrofill.js [hunk 2]
```diff
@@ -92,20 +92,24 @@ async function getUserAgentUsingClientHints(hints) {
...
   let is_chromium = false;
   let is_edge = false
   let chromium_version;
   navigator.userAgentData.brands.forEach(value => {
     if (value.brand == 'Chromium') {
-      is_chromium = true;
+      // Let's double check the UA string as well, so we don't accidentally
+      // capture a headless browser or friendly bot (which should report as
+      // HeadlessChrome or something entirely different).
+      is_chromium = true && /\sChrome/.test(navigator.userAgent);
```
Comment (miketaylr) [inline uach-retrofill.js:104]: OK, in theory this is what I originally intended to do.

(But there's no way to really know)

Comment (rowan-m) [reply uach-retrofill.js:104]: Can't the `true && ` just drop completely?

Comment (miketaylr) [reply uach-retrofill.js]: Heh, yes. Thank you.

**Comment by miketaylr** [comment]
Third time's a charm. PTAL @rowan-m

**Review by rowan-m — APPROVED** [review]

**Pull request merged**